src/adu_workflow/src/agent_workflow.c

```diff
@@ -1169,14 +1168,21 @@ static void CallDownloadHandlerOnUpdateWorkflowCompleted(const ADUC_WorkflowHand
     for (size_t i = 0; i < payloadCount; ++i)
     {
         ADUC_Result result = {};
-        ADUC_FileEntity* fileEntity = NULL;
-        if (!workflow_get_update_file(workflowHandle, i, &fileEntity) || IsNullOrEmpty(fileEntity->DownloadHandlerId))
+        ADUC_FileEntity fileEntity;
+        memset(&fileEntity, 0, sizeof(fileEntity));
```

Comment:
nit: zero not required.  

`foo f;
if (get_f(&f()) { ... }`

doesn't access f on failure, and should assume that get_f has fully intited f.

Reply:
(throughout)

Reply:
see my note below about when we *do* need the zero init.

Also, consider (not essential) changing the instances of memset() to:

    ADUC_FileEntity_Init(&fileEntity);

which just does a zero init currently.  It matches the Uninit call.

Reply:
workflow_get_update_file already calls ADUC_FileEntity_init on caller's behalf (line 1808), setting out param as a side-effect.